patches/tModLoader/Terraria/NPC.TML.cs

```diff
@@ -197,4 +197,52 @@ public int GetPartyHatGore()
 		}
 		return num;
 	}
+
+	/// <summary>
+	/// Adjusts <see cref="buffImmune"/> to make this NPC immune to the provided buff as well as all other buffs that inherit the immunity of that buff (via <see cref="BuffID.Sets.GrantImmunityWith"/>). This method can be followed by <see cref="ClearImmuneToBuffs(out bool)"/> if the NPC should clear any buff it currently has that it is now immune to.
+	/// </summary>
+	/// <param name="buffType"></param>
+	public void BecomeImmuneTo(int buffType)
+	{
+		buffImmune[buffType] = true; 
+
+		for (int i = 0; i < BuffID.Sets.GrantImmunityWith.Length; i++) {
+			var buffsToInherit = BuffID.Sets.GrantImmunityWith[i];
+			if (buffsToInherit.Contains(buffType)) {
```

Comment:
Could this be sped up perhaps? Or would that be premature optimization, since its unlikely that it will be called every tick, assuming modders follow the ExampleMod example.

Reply:
I'll make a note, but probably not needed for now.